docs/Practices/hub/hub.md

```diff
@@ -0,0 +1,116 @@
+# Working with Bounded Contexts in Hub
+
+## Intro
+This document defines a number of guidelines to follow while working in Hub. Guidelines for working across projects can be found here (*not yet written*).
+
+These guidelines are intended to lead an engineer towards achieving a set of goals outlined below. These guidelines are based around a set of [useful abstractions](../../glossary.md#useful-abstraction) (UAs).
+
+So what are some of the useful abstractions that we utilize? Some of the notable UAs are Domain Driven Design, Bounded Contexts, Command Query Separation, Verticle Slices Architecture, Functional styled Procedural Programming, Ports and Adapters, The Repository Pattern, and Onion (aka Clean aka Hexagonal) Architecture. Not all UAs are implemented dogmatically, and doing so would lead to many instances of conflict between the UAs. Rather choice core parts from different UAs have been chosen and applied in such a way that these parts are orthogonal to each other and support well the defined goals.
+
+
+This document will not attempt to answer all critiques of a given guideline or UA nor will it discuss in depth the tradeoffs of why these guidelines the associated UA(s) are the right tools.
+
+For more on different UAs check out the [learning doc](learning.md).
+
+## Goals
+- Code for a given capability is easy to find.
+- Changes to a given behavior are less risky.
+- The effects of a change to a domain model shape are clear and should cascade and only to those dependents that actually utilize the changes.
+- The API of code implementing a behavior is clear. *You should know exactly what data is required to successfully initiate a behavior as well as understanding the expected failures of that behavior*
+- Failures due to essential complexity should be uniquely and easily identifiable and bad data states caused by such errors should be easily remediated.
+- Code should be covered by automated tests. *Testing strategy and acceptable coverage are not currently specified, but shoot for as much as possible.*
+- Flexibility for implementation within defined bounraries. *Unlimited flexibility (autonomy) leads to multiple competing standards, analysis paralysis, lack of direction, and a lack of predictability in code. Inflexibility (too many conventions/only a single way to do something) leads to brittle code, overly complicated code for the purpose of adhering to defined rules, difficulty with corner cases. The sweet spot is one where there are multiple correct implementations that each can be derived from a set of UAs or guidelines. This provides confidence to an engineer because they are able to orient on a UA or guideline for direction while having to autonomy to being able to chart the appropriate course to get there.*
```

Comment:
```suggestion
- Flexibility for implementation within defined boundaries. *Unlimited flexibility (autonomy) leads to multiple competing standards, analysis paralysis, lack of direction, and a lack of predictability in code. Inflexibility (too many conventions/only a single way to do something) leads to brittle code, overly complicated code for the purpose of adhering to defined rules, difficulty with corner cases. The sweet spot is one where there are multiple correct implementations that each can be derived from a set of UAs or guidelines. This provides confidence to an engineer because they are able to orient on a UA or guideline for direction while having to autonomy to being able to chart the appropriate course to get there.*
```
```suggestion
- Flexibility for implementation within defined bounraries. *Unlimited flexibility (autonomy) leads to multiple competing standards, analysis paralysis, lack of direction, and a lack of predictability in code. Inflexibility (too many conventions/only a single way to do something) leads to brittle code, overly complicated code for the purpose of adhering to defined rules, difficulty with corner cases. The sweet spot is one where there are multiple correct implementations that each can be derived from a set of UAs or guidelines. This provides confidence to an engineer because they are able to orient on a UA or guideline for direction while having to autonomy to being able to chart the appropriate course to get there.*
```

Reply:
"while having to autonomy to being able to chart the appropriate course"

Something doesn't seem right in this sentence. Do you mean while having "the autonomy"?